src/IconWithLabel/IconWithLabel.tsx

```diff
@@ -33,13 +43,15 @@ export default function IconWithLabel({
         gap: "4px"
       }}
     >
-      <Box
-        sx={{
-          display: "flex"
-        }}
-      >
-        {customizedIcon}
-      </Box>
+      <Tooltip data-testid="icon-tooltip" title={tooltip}>
+        <Box
+          sx={{
+            display: "flex"
+          }}
+        >
+          {customizedIcon}
+        </Box>
+      </Tooltip>
```

Comment:
I would move the <Box> to its own component like:
```ts
const Icon = ({children}) => (<Box sx={{display: "flex"}}>{children}</Box>)
```
And then use this Icon in the following way:

```suggestion
      tooltip ? (<Tooltip data-testid="icon-tooltip" title={tooltip}>
        <Icon>{customizedIcon}</Icon>
      </Tooltip>) : <Icon>{customizedIcon}</Icon>
```

I am making the following suggestion as I don't think it makes sense from a testing perspective to have the data-testid "icon-tooltip" in the DOM when in the case when we are not using a tooltip.

Reply:
I have just rendered conditionally the icon with and without Tooltip as tooltip required when rendered functional component inside to be a forward ref and name as well. I just thought it's more code instead of simple ternary.